internal/infrastructure/mongo/mongodoc/pagination.go

```diff
@@ -0,0 +1,52 @@
+package mongodoc
+
+import (
+	"errors"
+
+	"github.com/reearth/reearth-backend/internal/usecase"
+)
+
+type Pagination struct {
+	Before *string
+	After  *string
+	First  *int
+	Last   *int
+}
```

Comment:
✅  Excellent!
Understanding that pagination for MongoDB is different from pagination for the use case layer, and separating them into separate structs to achieve separation of interest is a good point worth mentioning.